Zend/zend_language_parser.y

```diff
@@ -1301,10 +1301,6 @@ new_variable:
 			{ $$ = zend_ast_create_ex(ZEND_AST_DIM, ZEND_DIM_ALTERNATIVE_SYNTAX, $1, $3); }
 	|	new_variable T_OBJECT_OPERATOR property_name
 			{ $$ = zend_ast_create(ZEND_AST_PROP, $1, $3); }
-	|	new_variable T_NULLSAFE_OBJECT_OPERATOR property_name {
-			/* New object can never be null, we support this for consistency */
-			$$ = zend_ast_create(ZEND_AST_PROP, $1, $3);
```

Comment:
Shouldn't the parser rule stay there, just create AST_NULLSAFE_PROP?

Reply:
Sorry, my brain wasn't working yesterday. Note that using `?->` in `new` is pretty useless unless we short-circuit the object instantiation. Same with `clone`. I've suggested adding this but nobody seemed particularly convinced this is useful. Good thing is we'll always be able to do that with no BC breaks.

Reply:
I added a test for this (with no short-circuiting as suggested by the RFC). As mentioned, we could always add short-circuiting here later if we see fit.